surveys/2022-wg-debugging-survey/questions.md

```diff
@@ -0,0 +1,361 @@
+- I am uncomfortable using a debugger [`NEXT`](#why-are-you-uncomfortable-using-a-debugger)
+- I am comfortable using a debugger
+- I am very comfortable using a debugger
+
+> *Justification:*
+>
+> Users who are more familiar with debuggers will face different problems than
+> users who are not.
+
+### Why are you uncomfortable using a debugger?
+Type: select all that apply
+
+- Unfamiliar (just need to spend more time)
+- Reliability (don't expect debugger to work well enough)
+- Ease-of-use
+- Lack of documentation with Rust
+
+> *Justification:*
+>
+> Why aren't users comfortable with debuggers is useful.
+
+### Do you often use a debugger with other programming languages (anything but Rust)?
+Type: select one
+
+- Daily or nearly so
+- Weekly or nearly so
+- Monthly or nearly so
+- Rarely
+
+> *Justification:*
+>
+> If a user regularly uses a debugger for other languages but not for Rust then
+> it would be interesting to know that.
+
+### Which programming language(s) do you regularly use a debugger with?
+Type: select all that apply
+
+- C
+- C++
+- Java
+- JavaScript
+- Python
+- Go
+- C#
+- Other
+
+> *Justification:*
+>
+> Debugging experiences vary widely by language - debugging a interpreted
+> language like Python is very different from a compiled language like C/C++ -
+> this will inform user expectations with Rust debuggers.
+
+### How does debugging in language X compare to Rust?
+Type: freeform text
+
+> *Justification:*
+>
+> What can Rust's debugging experience learn from other languages.
+
+### Do you often use a debugger with Rust?
+Type: select one
+
+- Daily or nearly so
+- Weekly or nearly so
+- Monthly or nearly so
+- Rarely
+
+> *Justification:*
+>
+> Knowing how often a debugger is used with Rust is valuable, but need to know
+> whether that is because of the quality of the experience or because it
+> is/isn't often needed.
+
+### How do you start a debugger (when debugging Rust)?
+Type: select all that apply
+
+- Core dumps
+- Attaching to a process
+- Launching executable
+- From an IDE
+
+> *Justification:*
+>
+> User experience for starting a debugger in either of these ways could be
+> improved if they were particularly common or uncommon.
+
+### What debuggers have you heard of?
+Type: select all that apply
+
+- `gdb`
+- `lldb`
+- Visual Studio
+- WinDBG
+- `rr`
+
+> *Justification:*
+>
+> If users aren't aware of certain types of debugger than raising awareness or
+> making it easier to start these debuggers might be valuable.
+
+### What debuggers do you use with Rust?
+Type: select all that apply
+
+- `gdb`
+- `lldb`
+- Visual Studio
+- WinDbg
+- `rr`
+
+> *Justification:*
+>
+> Debuggers that are actually used is useful.
+
+### Which debugger interfaces do you use with Rust?
+Type: select all that apply
+
+- Editor extension (e.g. `termdebug`)
+- Built-in IDE debugger
+- Terminal line interface (e.g. `gdb`)
+- Terminal UI (e.g. `gdb`'s TUI)
+- Web interface (e.g. Pernosco)
+- Separate debugging program
```

Comment:
What does this mean?

Reply:
Changed this to "Other" because I wasn't sure either :)